examples/settings/settings_main.c

```diff
@@ -0,0 +1,276 @@
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/mount.h>
+
+#include "system/settings.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * settings_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  int                  ret;
+  int                  fd;
+  enum settings_type_e stype;
+  char                 path[50];
+  char                 *teststr = "";
+  int                  testval = 0x5aa5;
+  char                 readstr[CONFIG_SETTINGS_VALUE_SIZE];
+  enum storage_type_e  storage_type = STORAGE_BINARY;
+  bool                 flag_present = false;
+  struct stat          sbuf;
+
+  ASSERT(0);
+  if ((argc < 1) || *argv[1] == 0 || *(argv[1] + 1) == 0)
+    {
+      goto print_help;
+    }
+
+  if (*argv[1] == '-')
+    {
+      flag_present = true;
+      if (*(argv[1] + 1) == 'b')
+        {
+          storage_type = STORAGE_BINARY;
+        }
+      else if (*(argv[1] + 1) == 't')
+        {
+          storage_type = STORAGE_TEXT;
+        }
+      else
+        {
+          goto print_help;
+        }
+    }
+
+  if (flag_present && argc < 2)
+    {
+      goto print_help;
+    }
+
+#ifdef CONFIG_EXAMPLES_SETTINGS_USE_TMPFS
+#  ifndef CONFIG_FS_TMPFS
+#    error TMPFS must be enabled
+#  endif
+#  ifndef CONFIG_LIBC_TMPDIR
+#    error LIBC_TMPDIR must be defined
+#  endif
+  if (stat(CONFIG_LIBC_TMPDIR, &sbuf))
+    {
+      ret = nx_mount(NULL, CONFIG_LIBC_TMPDIR, "tmpfs", 0, NULL);
+      if (ret < 0)
+        {
+          printf("ERROR: Failed to mount tmpfs at %s: %d\n",
+                  CONFIG_LIBC_TMPDIR, ret);
+          goto end;
+        }
+    }
+
+  strcpy(path, CONFIG_LIBC_TMPDIR);
+  strcat(path, "/settings");
+#else
+  strcpy(path, CONFIG_EXAMPLES_SETTINGS_EXISTING_STORAGE);
```

Comment:
I would greatly prefer to use `char *path` and a `strdup()` instead of hard-coding the size of the path to 50.

If you really want to do it like this, maybe use `strncpy()`? To ensure that a long path will not cause an overflow.